spec/models/integration/post_show_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show page', type: :feature do
+  let(:user_name) { 'John Doe' }
+  let(:post_title) { 'Test Post Title' }
+  let(:post_body) { 'This is the post body text.' }
+
+  before do
+    @user = User.create(name: user_name)
+    @post = Post.create(title: post_title, text: post_body, author: @user)
+  end
+
+  it "displays the post's title" do
+    visit post_path(@post)
+    expect(page).to have_content(post_title)
+  end
+
+  it 'displays the number of comments' do
+    # Assuming you have comments_counter set up properly
+    @post.update(comments_counter: 1)
+    visit post_path(@post)
+    expect(page).to have_content('Comments: 1')
+  end
+
+  it 'displays the number of likes' do
+    # Assuming you have likes_counter set up properly
+    @post.update(likes_counter: 0)
+    visit post_path(@post)
+    expect(page).to have_content('Likes: 0')
+  end
+
+  it 'displays the post body' do
+    visit post_path(@post)
+    expect(page).to have_content(post_body)
+  end
+
+  it 'displays the username of each commentator' do
+    Comment.create(author: @user, post: @post, text: 'This is a comment.') # Use author: @user
+    visit post_path(@post)
+    expect(page).to have_content("Comment author: #{user_name}")
+  end
+
+  it 'displays the comment each commentor left' do
+    comment_text = 'This is a comment.'
+    Comment.create(author: @user, post: @post, text: comment_text)
+    visit post_path(@post)
+    expect(page).to have_content("Comment author: #{user_name}")
+    expect(page).to have_content("Comment content: #{comment_text}")
+  end
+end
```

Comment:
- [ ] Kindly complete the remaining test here too.

### Project requirements.
![post-show-rqt](https://github.com/theresetuyi/Blog-app/assets/26597432/1c2f3f43-e0e5-44f0-9541-dfeb80d07b9a)